include/zephyr/drivers/mfd/ad559x.h

```diff
@@ -78,8 +79,43 @@ int mfd_ad5592_read_reg(const struct device *dev, uint8_t reg, uint8_t reg_data,
  * @retval 0 if success
  * @retval negative errno if failure
  */
-int mfd_ad5592_write_reg(const struct device *dev, uint8_t reg, uint16_t val);
+int mfd_ad559x_write_reg(const struct device *dev, uint8_t reg, uint16_t val);
+
+/**
+ * @brief Read ADC channel data from the chip
+ *
+ * @param[in] dev Pointer to MFD device
+ * @param[in] channel Channel to read
+ * @param[out] result ADC channel value read
+ *
+ * @retval 0 if success
+ * @retval negative errno if failure
+ */
+int mfd_ad559x_read_adc_chan(const struct device *dev, uint8_t channel, uint16_t *result);
```

Comment:
I am open for suggestions!
I think I considered IC specific code in mfd_ad559x.c but it does not seem scalable.
I think I searched the code base for ideas, but I cannot remember if there was a driver that did something similar.

Reply:
Looking at the datasheet this is quite tricky for this device.
Most devices that support both I2C and SPI have near identical command sequences, so the difference are small, but for these chips the commands sequences are very different.

Could this be better handled by implementing the ad5593 as a separate driver instead?

Reply:
Take a look at https://github.com/zephyrproject-rtos/zephyr/tree/main/drivers/sensor/ina23x drivers. I know there is no MFD subsys used but we share the same codebase(not much but...) for both of them. Maybe AD5592 could be adapted similarly.

Reply:
@bbilas @aasinclair  I've looked at it for quite a bit now to figure out how to do it without adding either e.g. adc to mfd or mfd to adc.
The best solution I can come up with is to add a `AD559X_HAS_POINTER_BYTE_MAP` define in mfd and set it based on the bus type.
Then either have a `adc_ad559x_pbm.c` and a `adc_ad559x_no_pbm.c` files which has implement the differences or have `#ifdef` inside `adc_ad559x.c`.
I also considered adding `adc_ad5592.c` and ad5592 dts files, but that approach looks like it would require quite a bit of duplicated code in the `adc_ad5592.c` and `adc_ad5593` and potentially future ad559x ICs.

Reply:
@jeppenodgaard Please prepare some draft then we will be able to review it.